zmon-controller-app/src/main/java/org/zalando/zmon/service/impl/FalsePositiveRateServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+    with individual errors inside the response body:
+    https://opensource.zalando.com/restful-api-guidelines/#152 */
+    private static final Set<Integer> allowedStatusCode = ImmutableSet.of(200, 207);
+
+
+    @Autowired
+    public FalsePositiveRateServiceImpl(MetaDataProperties metaDataProperties,
+                                        AccessTokens accessTokens,
+                                        ObjectMapper mapper) {
+        this.metaDataProperties = metaDataProperties;
+        this.mapper = mapper;
+        this.executor = Executor.newInstance(metaDataProperties.getHttpClient());
+        this.accessTokens = accessTokens;
+    }
+
+    @Override
+    public ResponseEntity<JsonNode> getFalsePositiveRate(final String alertId) {
+        final String url = metaDataProperties.getUrl() + FALSE_POSITIVE_RATE_END_POINT + alertId;
+        try {
+            Request request = Request.Get(url);
+
+            log.debug("False positive alert id: {}", alertId);
+            log.debug("URL: {}", url);
+
+            request.addHeader(AUTHORIZATION, BEARER + accessTokens.get(ZMON_TOKEN_ID));
+            HttpResponse response = executor.execute(request).returnResponse();
+            return toResponseEntity(response);
+        } catch (Exception ex) {
+            log.error("Get false positive rate for Alert: {} failed", alertId, ex);
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public ResponseEntity<JsonNode> getFalsePositiveRateDataPoints(final String alertId, final Map<String, String> query) {
+        try {
+            String from = query.containsKey("from") ? URLEncoder.encode(query.get("from"), "UTF-8") : "";
+            String to = query.containsKey("to") ? URLEncoder.encode(query.get("to"), "UTF-8") : "";
+
+            log.info("Searching false-positive rate data points for Alert: {} From={} To={}", alertId, from, to);
+
+            UriComponentsBuilder urlBuilder = UriComponentsBuilder.fromUriString(metaDataProperties.getUrl())
+                    .path(FALSE_POSITIVE_RATE_END_POINT + alertId + "/datapoints")
+                    .queryParam("from", from)
+                    .queryParam("to", to);
+
+            log.debug("URL: {}", urlBuilder.build().toUri().toString());
+
+
+            Request request = Request.Get(urlBuilder.build().toUri());
+            request.addHeader(AUTHORIZATION, BEARER + accessTokens.get(ZMON_TOKEN_ID));
+            HttpResponse response = executor.execute(request).returnResponse();
+
+            return toResponseEntity(response);
+        } catch (Exception ex) {
+            log.error("Get false positive rate for Alert: {} failed", alertId, ex);
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public ResponseEntity<JsonNode> listFalsePositiveRates(final String alertIdList) {
+        log.info("Bulk get false positive rates");
```

Comment:
Does it really should have an INFO log level? What useful information it brings, if it doesn't log any variables?